tasks/publish-to-cgw/publish-to-cgw.yaml

```diff
@@ -0,0 +1,175 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: publish-to-cgw
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to publish content to CGW (Content-Gateway)
+  params:
+    - name: cgwHostname
+      type: string
+      description: >
+        The hostname of the content-gateway to publish the metadata to
+      default: https://developers.redhat.com/content-gateway/rest/admin
+    - name: cgwSecret
+      type: string
+      description: >
+        The kubernetes secret to use to authenticate to content-gateway.
+        It needs to contain two keys: username and token
+      default: publish-to-cgw-secret
+    - name: dataPath
+      type: string
+      description: >
+        Path to the JSON string of the merged data to use in the data workspace.
+    - name: contentDir
+      type: string
+      description: Path where the content to push is stored in the workspace
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used
+      type: string
+      default: ""
+  results:
+    - name: resultDataPath
+      type: string
+      description: |
+        The relative path in the workspace to the stored json of result data of the task
+  workspaces:
+    - name: data
+      description: Workspace to save the CR jsons to
+  steps:
+    - name: run-push-cgw-metadata
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+      env:
+        - name: CGW_USERNAME
+          valueFrom:
+            secretKeyRef:
+              name: $(params.cgwSecret)
+              key: username
+        - name: CGW_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.cgwSecret)
+              key: token
+      script: |
+        #!/usr/bin/env bash
+        python3 <<EOF
+        """
+        Run `push-cgw-metadata` command to push metadata to CGW:
+        1. Extract all components under contentGateway key from dataPath
+        2. Find all the files in contentDir that starts with the component name
+        4. Generate necessary metadata for each file
+        5. Dump the metadata to a YAML file
+        6. Run `push-cgw-metadata` to push the metadata
+        """
+        import os
+        import json
+        import yaml
+        import hashlib
+        import subprocess
+
+        DATA_FILE = "$(workspaces.data.path)/$(params.dataPath)"
+        CONTENT_DIR = "$(workspaces.data.path)/$(params.contentDir)"
+        WORKSPACE_DIR="$(workspaces.data.path)/$(params.subdirectory)"
```

Comment:
This is going to collide on parallel runs. You likely want to `dirname` the data file or something. We store everything in a subdirectory (check out collect-data if you want to see how it is done)

Reply:
make sense, I updated both variables to use  `WORKSPACE_DIR` which has already has `$(params.subdirectory)`